craft_application/services/remotebuild.py

```diff
@@ -325,7 +339,8 @@ def _new_recipe(
     ) -> launchpad.models.Recipe:
         """Create a new recipe for the given repository."""
         repository.lp_refresh()  # Prevents a race condition on new repositories.
-        git_ref = parse.urlparse(str(repository.git_https_url)).path + "/+ref/main"
+        url = repository.git_ssh_url if repository.private else repository.git_https_url
```

Comment:
I gather private repos can't be pushed over HTTPS?

Do you think we could get away with defaulting to ssh URLs everywhere?

Reply:
Add a comment about backward compat

Reply:
Added.